docs/src/modules/components/AppFrame.js

```diff
@@ -51,6 +50,29 @@ Router.onRouteChangeError = () => {
   NProgress.done();
 };
 
+const AppSearch = React.lazy(() => import('docs/src/modules/components/AppSearch'));
+function DeferredAppSearch() {
+  const fallback = null;
+
+  return (
+    <React.Fragment>
+      <link
+        rel="preload"
```

Comment:
Should we use prefetch to reduce the priority? https://3perf.com/blog/link-rels/

Reply:
As far as I can tell 

> It’s helpful when you need that resource a few seconds after loading the page, and you want to speed it up.

fits exactly how we're using it currently.

Reply:
I was asking because it might delay another resource, a resource that might be more important to get to the interactive state, previously we were waiting for the interactive state to load the CSS asset. Now, I haven't checked the actual implication, It's only a guess.

Reply:
I don't know what I'm supposed to do with guesses in a code review. The documentation you linked supports the implementation. If there are more important resources blocked by this implementation it would help if you include what these resources are and why you think they are more important. 

Just arguing against an implementation with guesses is not helpful.

Reply:
It was an invitation to explore this path. It seems to be an issue: https://www.webpagetest.org/video/compare.php?tests=190917_YM_74b7ddcba6319bbfca4d7c222ea68345,190917_1H_0ff4283726ec6cfc83b0f994b87f446b. We load the assets before the interactive state.

Reply:
What do you think of removing the preload link?

Reply:
Why do you want to remove it?

Reply:
The CSS asset is now loaded before the "time to interactive" event. My concern is with the potential usage of the network resource to load this asset instead of loading, let's say, JavaScript assets to get to the "time to interactive" event. I'm wondering if it can be a bottleneck, by having too much priority.

If you think that this concern doesn't hold in practice, I'm good.

Reply:
I can't identify any issue in the link. Could you point to the specific issue? As far as I can tell the graphs are similar.

Reply:
Sure, here is the difference I could notice:

**branch**
![branch](https://user-images.githubusercontent.com/3165635/65259238-a7053800-db04-11e9-974b-b21a9f0792f8.png)

**master**
![master](https://user-images.githubusercontent.com/3165635/65259241-a7053800-db04-11e9-84ca-71ff729ce9c0.png)

Reply:
No strong preference either.

Reply:
But why would you care about the order? The user won't notice. What's important is are user metrics like time to first byte or first meaningful paint or time to interactive. None of these are affected in any meaningful way.

You're looking for any difference and calling it a regression which is obvious since something was changed. What's important is the difference for the user not some theoretical internal difference, no?

Even more so why would you think it's ok for a stylesheet to be loaded after time to interactive? It can display search results without styling.

Reply:
The difference is only noticeable on a slow connection: https://www.webpagetest.org/video/compare.php?tests=190920_BS_a465ecd007173c07ecbd9e31fc2e1cf4,190920_D1_023f612a117b19cdd554390950f98104. Assuming we have to prioritize the order in which we load the assets, I believe that the search one are less important because the search to be displayed requires the following steps:
1. The page is interactive
1. The Algolia JavaScript asset is loaded
1. The search is visible
1. The user starts a search

We should be able to get the CSS loaded at 2, before 4 triggers.

Reply:
Anyway, it's not very important, move forward as you see fit!

Reply:
I wouldn't understand why an asset with preload would block either the suspended chunk or the page becoming interactive. The whole point of this is to signal to the browser: "no biggy if you don't load this but when you have the time start loading this please". I wouldn't bother too much looking into order of the loads because this changes by nature. You should not bother about the order because a dedicated scheduler does that.

There are too many moving parts here that you can't be sure this is a bad change. There are even some metrics that are better with this change. Another issue is that the time to first byte is already slower in the preload benchmark which surely is just random error that already gives the master benchmark a head start. 

Like this is no easy graph to read. You can't just look at two numbers and go: yep that's worse. As I read it this change makes the first interaction possible after 8s down from 9s (red is non-interactive because the main thread is blocked):
master:
![Screenshot from 2019-09-20 17-39-59](https://user-images.githubusercontent.com/12292047/65340068-b2706600-dbcd-11e9-813c-15b2937a44e7.png)

preload:
![Screenshot from 2019-09-20 17-39-24](https://user-images.githubusercontent.com/12292047/65340047-a684a400-dbcd-11e9-9128-9b1415fc76b2.png)